### PR TITLE
Handle idempotenacy in delete operation and convert variable in snakecase convention and update documentation as well.

### PR DESCRIPTION
**Pull request by Abhishek-121**
- Handle Idempotenacy in delete operation for site module if we run the same playbook again or if site not found.
- Convert the variables in the playbook and module documentation into snake-case to maintain uniformity across the modules.

### CODE REVIEW

File: plugins/modules/site_intent.py [hunk 1]
```diff
@@ -135,6 +154,68 @@
...
+    dnac_port: "{{dnac_port}}"
+    dnac_version: "{{dnac_version}}"
+    dnac_debug: "{{dnac_debug}}"
+    dnac_log: "{{dnac_log}}"
+    state: merged
+    config:
+    - site:
+        floor:
+          name: string
+          parentName: string
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:173]: Don't we need to change this with _ ?

File: plugins/modules/site_intent.py [hunk 2]
```diff
@@ -135,6 +154,68 @@
...
+          length: int
+          width: int
+          height: int
+          rfModel: string
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:177]: Don't we need to change this with _ ?

File: plugins/modules/site_intent.py [hunk 3]
```diff
@@ -135,6 +154,68 @@
...
+          floorNumber: int
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:178]: Don't we need to change this with _ ?

File: plugins/modules/site_intent.py [hunk 4]
```diff
@@ -135,6 +154,68 @@
...
+    dnac_port: "{{dnac_port}}"
+    dnac_version: "{{dnac_version}}"
+    dnac_debug: "{{dnac_debug}}"
+    dnac_log: "{{dnac_log}}"
+    state: merged
+    config:
+    - site:
+        floor:
+          name: string
+          parentName: string
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:196]: Don't we need to change this with _ ?

File: plugins/modules/site_intent.py [hunk 5]
```diff
@@ -135,6 +154,68 @@
...
+    dnac_port: "{{dnac_port}}"
+    dnac_version: "{{dnac_version}}"
+    dnac_debug: "{{dnac_debug}}"
+    dnac_log: "{{dnac_log}}"
+    state: deleted
+    config:
+    - site:
+        floor:
+          name: string
+          parentName: string
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:217]: Don't we need to change this with _ ?

File: plugins/modules/site_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:173]: somehow missed it in the documentation.

File: plugins/modules/site_intent.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:177]: changed

File: plugins/modules/site_intent.py (hunk 4 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:196]: changed

File: plugins/modules/site_intent.py (hunk 5 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:217]: changed

File: plugins/modules/site_intent.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:178]: changed

File: plugins/modules/site_intent.py [hunk 6]
```diff
@@ -690,7 +797,12 @@ def get_diff_deleted(self, config):
                         break
 
         else:
-            self.module.fail_json(msg="Site Not Found", response=[])
+            msg = "Cannot delete Site - {0} as it's not found in Cisco DNA Center".format(self.want.get("site_name"))
+            self.status = "success"
+            self.result['changed'] = False
+            self.result['response'] = msg
+            log(msg)
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:804]: can you change to self.log?

**Review by madhansansel — APPROVED** [review]

**Pull request merged**